roles/infrastructure/tasks/setup_azure_storage.yml

```diff
@@ -32,6 +34,23 @@
       location:  "{{ infra__region }}"
       tags: "{{ infra__tags }}"
 
+- name: Check Storage Account creation result for failure
+  when:
+    - __infra_az_stor_acccount_test.rc is defined
+    - __infra_az_stor_acccount_test.rc == 1
+  block:
+    - name: Test Azure Storage Account creation for StorageAccountAlreadyTaken
```

Comment:
We should do this for AWS too. Many a time I have torn down without deleting data. And I get a strange AWS error. 

I realize I just volunteered to do this...

Reply:
@wmudge suggested we put this into the AWS side of things, and even suggested the approach I have used here, so it would be good to cross pollenate as a pattern where we know a common failure happens and we want to give a very specific instruction to the user to avoid confusion.